appwrite/appwrite
# Fix bucket permissions

## What does this PR do?

The metadata collection was not migrated so it still had `["read(\"any\")","create(\"any\")","update(\"any\")","delete(\"any\")"]` permissions, giving everyone read access.

In addition, the create and update bucket APIs did not update the metadata collection so the permissions were out of sync.

## Test Plan

Manual:

1. Deployed 1.2.1
2. Create a bucket with file security = true
3. Upload a file
4. Ensure a user doesn't have access to the file
5. Upgrade & migrate
6. Ensure a user doesn't have access to the file
7. Update bucket permissions to allow any to read
8. Ensure a user does have access to the file
9. Update bucket permissions to remove any access to the file
10. Ensure a user doesn't have access to the file

Database after migrate:

<img width="777" alt="image" src="https://user-images.githubusercontent.com/1477010/236026458-51b33e9c-a27b-49c2-8614-21d05fcaf194.png">

with file security = true, joe only has access to joe's files:

<img width="1430" alt="image" src="https://user-images.githubusercontent.com/1477010/236026626-54aa19f1-ed91-4a68-a5fd-5ef93d5f43be.png">

with file security = false, joe can't access the bucket:

<img width="1007" alt="image" src="https://user-images.githubusercontent.com/1477010/236027079-c2bba7ce-b25f-4c1c-8994-75da464439a3.png">

with collection level read access, joe can see all files:

<img width="1428" alt="image" src="https://user-images.githubusercontent.com/1477010/236027253-d3805c6e-8dbd-4f3e-930f-2927507b4fd3.png">

## Related PRs and Issues

None

## Checklist
